### PR TITLE
[System.Private.CoreLib] Implement more APIs

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: mcs/class/System.Private.CoreLib/System.Runtime.CompilerServices/RuntimeHelpers.cs [hunk 1]
```diff
@@ -1,7 +1,113 @@
...
+		{
+			return SufficientExecutionStack ();
+		}
+
+		public static void ExecuteCodeWithGuaranteedCleanup (TryCode code, CleanupCode backoutCode, Object userData)
+		{
+		}
+
+		public static void PrepareConstrainedRegions ()
+		{
```
Comment (filipnavara) [inline mcs/class/System.Private.CoreLib/System.Runtime.CompilerServices/RuntimeHelpers.cs]: It would be nice to have a comment that says that these methods are intentionally empty. Alternatively they could probably be moved to the shared CoreLib partition since the (empty) implementation will likely be identical across runtimes (https://github.com/dotnet/coreclr/issues/17903).

File: mcs/class/System.Private.CoreLib/System.Threading/EventWaitHandle.cs [hunk 2]
```diff
@@ -1,15 +1,78 @@
...
+
+			createdNew = true;
+		}
+
+		static OpenExistingResult OpenExistingWorker (string name, out EventWaitHandle result)
+		{
+			throw new PlatformNotSupportedException (SR.PlatformNotSupported_NamedSynchronizationPrimitives);
+		}
+
+		internal static bool Set (SafeWaitHandle waitHandle) => throw new NotImplementedException ();
```
Comment (filipnavara) [inline mcs/class/System.Private.CoreLib/System.Threading/EventWaitHandle.cs]: Would be nice to have this implemented, it is practically identical to the `Set` above.

File: mcs/class/System.Private.CoreLib/System.Runtime.CompilerServices/RuntimeHelpers.cs (hunk 1 above)
Comment (marek-safar) [reply mcs/class/System.Private.CoreLib/System.Runtime.CompilerServices/RuntimeHelpers.cs]: I have the shared PR unification somewhere but the coverage is quite low (so low priority for now)

File: mono/metadata/icall-def.h [hunk 3]
```diff
@@ -124,6 +124,11 @@ HANDLES(NATIVEMETHODS_9, "SetProcessWorkingSetSize", ves_icall_Microsoft_Win32_N
 HANDLES(NATIVEMETHODS_10, "TerminateProcess", ves_icall_Microsoft_Win32_NativeMethods_TerminateProcess, MonoBoolean, 2, (gpointer, gint32))
 HANDLES(NATIVEMETHODS_11, "WaitForInputIdle", ves_icall_Microsoft_Win32_NativeMethods_WaitForInputIdle, gint32, 2, (gpointer, gint32))
 
+#if ENABLE_NETCORE // && UNIX
```
Comment (akoeplinger) [inline mono/metadata/icall-def.h]: the comment seems unrelated

Comment (filipnavara) [reply mono/metadata/icall-def.h]: On non-Unix builds it P/Invokes directly to the Win32 API, so the comment makes sense.
